scheduler-interface-spec.md

```diff
@@ -566,6 +561,10 @@ message AllocationRelease {
   TerminationType terminationType = 4;
   // human-readable message
   string message = 5;
+  // AllocationKey from AllocationAsk
+  string allocationkey = 6;
+  // update cache, default is false
+  bool updateCache = 7;
```

Comment:
Can you explain why we would ever not want to update the cache. Would that not cause issues with an out of sync cache?
The cache should be idempotent and update when needed:
- delete can never fail: if the entry does not exist the result is the same as a successful delete
- for creates and updates we should do a create OR update call with the end result being a consistent cache representing the correct object.

Reply:
context#processAllocationReleases is the only place where ReSyncSchedulerCache plugin is used to call ForgetPod in shim side for cache updates in addition to sending RMReleaseAllocationEvent event. RMReleaseAllocationEvent uses AllocationRelease message. AllocationRelease message is being used in few more places (for ex, context#handleRMUpdateApplicationEvent, node decommission etc) in addition to  context#processAllocationReleases in core side. Hence, to help shim whether to do cache operation or not while processing UpdateAllocation callback method, used "updateCache" field